src/test/kotlin/sirius/db/mongo/MongoTest.kt

```diff
@@ -8,101 +8,128 @@
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
 
-class MongoSpec extends BaseSpecification {
-
-    @Part
-    private static Mongo mongo
-
-            @Part
-            private static KeyGenerator keyGen
-
-            def "basic read / write works"() {
-        given:
-        def testString = String.valueOf(System.currentTimeMillis())
-        when:
-        def result = mongo.insert().set("test", testString).set("id", keyGen.generateId()).into("test")
-        then:
-        mongo.find().
-        where("id", result.getString("id")).
-        singleIn("test").
-        map({ d -> d.getString("test") }).
-        orElse(null) == testString
+@ExtendWith(SiriusExtension::class)
+class MongoTest {
+    companion object {
+        @Part
+        private lateinit var mongo: Mongo
+
+        @Part
+        private lateinit var keyGen: KeyGenerator
     }
 
-    def "read from secondary works"() {
-        given:
-        def testString = String.valueOf(System.currentTimeMillis())
-        when:
-        def result = mongo.insert().set("test", testString).set("id", keyGen.generateId()).into("test")
-        then:
-        mongo.findInSecondary().
-        where("id", result.getString("id")).
-        singleIn("test").
-        map({ d -> d.getString("test") }).
-        orElse(null) == testString
+    @Test
+    fun `basic read and write works`() {
+        val testString = System.currentTimeMillis().toString()
+        val result = mongo.insert().set("test", testString).set("id", keyGen.generateId()).into("test")
+
+        assertEquals(
+                testString,
+                mongo.find().where("id", result.getString("id")).singleIn("test").map { d -> d.getString("test") }
+                        .orElse(null)
+        )
+    }
+
+    @Test
+    fun `read from secondary works`() {
+        val testString = System.currentTimeMillis().toString()
+        val result = mongo.insert().set("test", testString).set("id", keyGen.generateId()).into("test")
+
+        assertEquals(
+                testString,
+                mongo.findInSecondary().where("id", result.getString("id")).singleIn("test")
+                        .map { d -> d.getString("test") }
+                        .orElse(null)
+        )
     }
 
-    def "sort works for singleIn"() {
-        when:
-        def result1 = mongo.insert().set("sortBy", 1).set("id", keyGen.generateId()).into("test")
-        def result2 = mongo.insert().set("sortBy", 3).set("id", keyGen.generateId()).into("test")
-        def result3 = mongo.insert().set("sortBy", 2).set("id", keyGen.generateId()).into("test")
-        then:
-        mongo.find()
+    @Test
+    fun `sort works for singleIn`() {
+        mongo.insert().set("sortBy", 1).set("id", keyGen.generateId()).into("test")
+        mongo.insert().set("sortBy", 2).set("id", keyGen.generateId()).into("test")
+        val result2 = mongo.insert().set("sortBy", 3).set("id", keyGen.generateId()).into("test")
```

Comment:
Here the sortby values are flipped. Does this not alter the scenario of the test?